feat(documentation): add toast stories (#878)
swisspost/design-system: packages/documentation/src/stories/components/toast/toast.docs.mdx

<Meta title="Components/Toast" />

# Toast

<p className="lead">
  Push notifications to your visitors with a <code>.toast</code> element, a lightweight and easily
  customizable alert message.
</p>

Toasts are lightweight notifications designed to mimic the push notifications that have been popularized by mobile and desktop operating systems.
They are intended to be small interruptions to your visitors or users, and therefore should contain minimal, to-the-point, non-interactive content.

<Canvas>
  <Story id="components-toast--default" />
</Canvas>
<div className="hide-col-default">
  <ArgsTable story="^" />
</div>

## SCSS Imports

To import all styles from the design-system-styles package:

```css dark
@use '@swisspost/design-system-styles/index.scss';
```

To import only the styles required for this component:

```css dark
@use '@swisspost/design-system-styles/basics.scss';
@use '@swisspost/design-system-styles/components/toast.scss';
```

## Auto close

<Canvas>
  <Story id="components-toast--auto-close" />
</Canvas>

Toasts should be closed automatically after a while, because they usually overlay the regular content.

> Every change on the screen should be left for at least 6 seconds so that the user can perceive it.

Reviewer: ```suggestion
Each change on the screen should remain for at least 6 seconds for the user to perceive it.
```

> We therefore recommend a display duration of at least 10-12 seconds, since the message still has to be

Reviewer: ```suggestion
We therefore recommend a display time of at least 10-12 seconds, since the message still has to be
```

read.

> In addition, if the user moves the mouse or the focus to the `.toast` element, the automatic closing should be disabled. If the mouse/focus leaves the element again, the timeout for the automatic closing can be restarted.

Reviewer: ```suggestion
In addition, if the user moves the mouse or the focus to the `.toast` element, the automatic closing should be disabled. If the mouse/focus leaves the element again, the automatic closing delay should be restarted.
```

## Accessibility

> Toasts are intended to be small interruptions to your visitors or users, so to help those with screen readers and similar assistive technologies, you should wrap your toasts in an `aria-live` region. Changes to live regions (such as injecting/updating a toast component) are automatically announced by screen readers without needing to move the user’s focus or otherwise interrupt the user. Additionally, include `aria-atomic="true"` to ensure that the entire toast is always announced as a single (atomic) unit, rather than just announcing what was changed (which could lead to problems if you only update part of the toast’s content, or if displaying the same toast content at a later point in time). If the information needed is important for the process, e.g. for a list of errors in a form, then use the [alert component](/?path=/docs/components-alert--default) instead of toast.

Reviewer: ```suggestion
Toasts are intended to be small interruptions to your visitors or users, so to help those with screen readers and similar assistive technologies, you should wrap your toasts in an `aria-live` region. Changes to live regions (such as injecting/updating a toast component) are automatically announced by screen readers without the need to move the user’s focus or interrupt in any way. Additionally, include `aria-atomic="true"` to ensure that the entire toast is always announced as a single (atomic) unit, rather than just announcing what changed (which could lead to problems if you only update part of the toast’s content, or if you show the same toast content later in time). If the information is important in the process, e.g. for a list of errors in a form, then use the [alert component](/?path=/docs/components-alert--default) instead of the toast.
```

Note that the live region needs to be present in the markup before the toast is generated or updated. If you dynamically generate both at the same time and inject them into the page, they will generally not be announced by assistive technologies.

> You also need to adapt the `role` and `aria-live` level depending on the content. If it’s an important message like an error, use `role="alert" aria-live="assertive"`, otherwise use `role="status" aria-live="polite"` attributes.

Reviewer 2: Nice write up. Please note that ng-bootstrap already provides aria-live attributes on the container, so this is only relevant for users of the plain bootstrap toasts. For them, we could also link to the relevant bootstrap docs section (https://getbootstrap.com/docs/5.1/components/toasts/#stacking).

Author: Haha, you mean nice copy-paste 🤣

It was my intention, to use this Page only to document the bootstrap/toast component (at least for the moment).
Hope this is ok?

Would you prefer to just link over to bootstrap or to implement an stacking example by our own?

Reviewer 2: ![image](https://user-images.githubusercontent.com/1659006/207782291-5059f0d9-af1d-48d6-b5b4-bbfab69e4bdc.png)

Yes, documenting plain bootstrap only is ok. A stacking example would be nice though (stop disabling that button and let me hammer out as many toasts as I can).
